lib/site.js

```diff
@@ -73,10 +70,19 @@ module.exports = class Site {
 
 			page.filepath = filepath;
 
+			return page.load();
+		});
+
+		await Promise.all(preperations);
+
+		let layout = generateLayout(tree, this.assetManager.manifest);
```

Comment:
seems like this should be asynchronous, for extensibility

Reply:
Done.